src/Reflex/Host/Headless.hs

```diff
@@ -0,0 +1,85 @@
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE ConstraintKinds #-}
+
+module Reflex.Host.Headless where
+
+import Data.Maybe
+import Reflex
+import Control.Monad.Fix
+import Control.Monad.Primitive
+import Reflex.Host.Class
+import Control.Monad.IO.Class
+import Control.Monad.Ref
+import Data.IORef
+import Data.Dependent.Sum
+import Control.Concurrent.Chan (newChan, readChan)
+import Control.Monad (forM, forM_)
+import Control.Monad.Identity (Identity(..))
+
+type MonadHeadlessApp t m =
+  ( Reflex t
+  , MonadHold t m
+  , MonadFix m
+  , PrimMonad (HostFrame t)
+  , ReflexHost t
+  , MonadIO (HostFrame t)
+  , Ref m ~ IORef
+  , Ref (HostFrame t) ~ IORef
+  , MonadRef (HostFrame t)
+  , NotReady t m
+  , TriggerEvent t m
+  , PostBuild t m
+  , PerformEvent t m
+  , MonadIO m
+  , MonadIO (Performable m)
+  , Adjustable t m
+  )
+
+-- | Run a headless FRP network. Inside the action, you will most probably use
+-- the capabilities provided by the 'TriggerEvent' and 'PerformEvent' type class
+-- to interface the FRP network with the outside world. Useful for testing.
+runHeadlessApp
+  :: (forall t m. MonadHeadlessApp t m => m (Event t ()))
+  -- ^ The action to be run in the headless FRP network
```

Comment:
Why does this action return an Event?

Reply:
Because the occurrence of that event is used in the code to signal the shutdown of the network. What would the return type that you have in mind be?

Reply:
Can you document that?

Reply:
sure, added that as part of the haddock